quantize.py

```diff
@@ -414,15 +415,15 @@ def create_quantized_state_dict(self):
                             "and that groupsize and inner_k_tiles*16 evenly divide into it")
                         continue
                 weight_int4pack, scales_and_zeros = prepare_int4_weight_and_scales_and_zeros(
-                    weight.to(torch.bfloat16).to('cuda'), self.groupsize, self.inner_k_tiles
+                    weight.to(torch.bfloat16).to(device=self.device), self.groupsize, self.inner_k_tiles
```

Comment:
I've always been surprised by this .to() here?  Do we actually need this?  Would the weights not be on the target device already?  I'm sure I'm missing something here?